src/eleveldb_metadata.erl

```diff
@@ -0,0 +1,110 @@
+%% -------------------------------------------------------------------
+%%
+%%  eleveldb_metadata: Erlang interface to allow LevelDB to retrieve riak_core metadata
+%%
+%% Copyright (c) 2017 Basho Technologies, Inc. All Rights Reserved.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% -------------------------------------------------------------------
+-module(eleveldb_metadata).
+
```

Comment:
I realize this is designed to handle raw messages from a NIF, but a protocol interface with lots and lots of doc comments seems like a good thing to have.